packages/browser/src/profiling/utils.ts

```diff
@@ -195,6 +211,161 @@ export function createProfilePayload(
   return profile;
 }
 
+/**
+ * Create a profile chunk envelope item
+ */
+export function createProfileChunkPayload(
+  jsSelfProfile: JSSelfProfile,
+  client: Client,
+  profilerId?: string,
+): ProfileChunk {
+  // only == to catch null and undefined
+  if (jsSelfProfile == null) {
+    throw new TypeError(
+      `Cannot construct profiling event envelope without a valid profile. Got ${jsSelfProfile} instead.`,
+    );
+  }
+
+  const continuousProfile = convertToContinuousProfile(jsSelfProfile);
+
+  const options = client.getOptions();
+  const sdk = client.getSdkMetadata?.()?.sdk;
+
+  return {
+    chunk_id: uuid4(),
+    client_sdk: {
+      name: sdk?.name ?? 'sentry.javascript.browser',
+      version: sdk?.version ?? '0.0.0',
+    },
+    profiler_id: profilerId || uuid4(),
+    platform: 'javascript',
+    version: '2',
+    release: options.release ?? '',
+    environment: options.environment ?? 'production',
+    debug_meta: {
+      // function name obfuscation
+      images: applyDebugMetadata(jsSelfProfile.resources),
+    },
+    profile: continuousProfile,
+  };
+}
+
+/**
+ * Validate a profile chunk against the Sample Format V2 requirements.
+ * https://develop.sentry.dev/sdk/telemetry/profiles/sample-format-v2/
+ * - Presence of samples, stacks, frames
+ * - Required metadata fields
+ */
+export function validateProfileChunk(chunk: ProfileChunk): { valid: boolean; reason?: string } {
```

Comment:
bundle size suggestion (feel free to ignore): we can probably save some more bytes here by only returning `{ valid: true }` or `{reason: string}` (i.e. avoid returning `valid: false` a bunch of times).